File: sparkcql-core/src/main/scala/gov/va/sparkcql/translator/node/NullNode.scala
package gov.va.sparkcql.translator.node

import org.hl7.elm.{r1 => elm}
import gov.va.sparkcql.translator._

class NullNode() extends Node {

  val element: elm.Element = null

  override def translate(env: Environment): Object = {
    null
  }
}

File: transpiler/src/main/java/gov/va/transpiler/jinja/node/trackable/element/expression/NullNode.java
package gov.va.transpiler.jinja.node.trackable.element.expression;

import org.hl7.elm.r1.Null;

import gov.va.transpiler.jinja.state.State;

public class NullNode extends ExpressionNode<Null> {

    public NullNode(State state, Null cqlEquivalent) {
        super(state, cqlEquivalent);
    }

    @Override
    public int allowedNumberOfChildren() {
        return 0;
    }
}
